### PR TITLE
support uninstall

### LINKED ISSUE #12: support uninstall
Currently the base bundle only supports installation. Need to support uninstall that cleans up all the resources.

### PR DESCRIPTION
**Pull request by sbawaska**
While installing the bundle a label is applied to all the resources being installed. Uninstall then deletes the resources with this  label.

closes #12

### CODE REVIEW

File: pkg/kab/patch_manifest.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ const (
 	DOCKER_FOR_DESKTOP_NAME        = "docker-for-desktop"
 	NODE_PORT_ENV_VAR              = "NODE_PORT"
 	CNAB_INSTALLATION_NAME_ENV_VAR = "CNAB_INSTALLATION_NAME"
+	LABEL_KEY_NAME                 = "bundle-installation-name"
```
Comment (glyn) [inline pkg/kab/patch_manifest.go]: Does this need qualifying so we don't accidentally delete resources created by others?

File: pkg/kab/uninstall.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	return nil
+}
+
+func (c *Client) LookupManifest(name string) (*v1alpha1.Manifest, error) {
+	namespaceList, err := c.coreClient.CoreV1().Namespaces().List(v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, ns := range namespaceList.Items {
```
Comment (glyn) [inline pkg/kab/uninstall.go]: It seems that an installation is cluster-wide and assumed to be only in a single namespace. Is that the case?

File: pkg/kab/uninstall_test.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+				},
+			}, nil)
+			fakeKabClient.PrependReactor("get", "*", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+				return true, nil, errors.NewNotFound(schema.GroupResource{}, "none")
+			})
+			_, err = client.LookupManifest("myInstallation")
+			Expect(err).To(MatchError("could not find manifest for installation name: myInstallation"))
+		})
+	})
+	Context("When there are more than one namespaces", func() {
```
Comment (glyn) [inline pkg/kab/uninstall_test.go]: Typo.
```suggestion
	Context("When there is more than one namespaces", func() {
```

File: pkg/kab/uninstall_test.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+		It("all namespaces are looked up to find a manifest", func() {
```
Comment (glyn) [inline pkg/kab/uninstall_test.go]: What about the edge case where a manifest somehow exists in more than one namespace? Unless we can guarantee that this never happens (e.g. do we check during installation that the manifest isn't already present in another namespace?), we should test for it?

File: pkg/scan/list_kind.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	"sort"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/pivotal/go-ape/pkg/furl"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+)
+
+func ListKind(res string, baseDir string) ([]string, error) {
```
Comment (glyn) [inline pkg/scan/list_kind.go]: Where is this used, apart from in unit tests?

File: pkg/kab/patch_manifest.go (hunk 1 above)
Comment (sbawaska) [reply pkg/kab/patch_manifest.go]: @glyn I added the `bundle` prefix to qualify this label. I had also considered `cnab-k8s-installer-installation-name`, but chose the former. Would you prefer the later?

File: pkg/kab/uninstall.go (hunk 2 above)
Comment (sbawaska) [reply pkg/kab/uninstall.go]: Yes, the installation is cluster-wide but the manifest itself is stored in a namespace.

File: pkg/scan/list_kind.go (hunk 5 above)
Comment (sbawaska) [reply pkg/scan/list_kind.go]: At the moment it is a convenience method for the unit test, also maintains parity with ListImages.

File: pkg/kab/uninstall_test.go (hunk 4 above)
Comment (sbawaska) [reply pkg/kab/uninstall_test.go]: Yes, installation fails if it finds a manifest.

File: pkg/kab/patch_manifest.go (hunk 1 above)
Comment (glyn) [reply pkg/kab/patch_manifest.go]: Yes, I think that's safer.

File: pkg/kab/uninstall.go (hunk 2 above)
Comment (glyn) [reply pkg/kab/uninstall.go]: Ok, thanks.

File: pkg/kab/uninstall_test.go (hunk 4 above)
Comment (glyn) [reply pkg/kab/uninstall_test.go]: Great, thanks.

File: pkg/scan/list_kind.go (hunk 5 above)
Comment (glyn) [reply pkg/scan/list_kind.go]: Ok. Might be worth a comment to that effect.

**Pull request merged**